lua/acf/hooks/hooks_sv.lua

```diff
@@ -0,0 +1,239 @@
+	--- @param Entity entity The entity that will be using the ammo type.
+	--- @param Data table The table of entity information that was used on the entity.
+	--- @param ... any One or many tables or objects that are related to the entity, these will vary on every class.
+	function Gamemode:ACF_OnAmmoFirst()
+	end
+
+	--- Called when an ammo type object is about to be replaced by another one or just removed.
+	--- @param AmmoType table The ammo type object that will be replaced or removed.
+	--- @param Entity entity The entity containing the ammo type object.
+	function Gamemode:ACF_OnAmmoLast()
+	end
+
+	--- Called when an ammo crate attempts to replenish another one.
+	--- @param Refill entity The ammo crate that will be providing ammunition.
+	--- @param Crate entity The ammo crate that will get replenished.
+	--- @param Amount number The quantity of ammunition that will get replenished.
+	--- @return boolean # True if the crate can be replenished by the refill with the given amount, false otherwise.
+	function Gamemode:ACF_AmmoCanRefill()
+		return true
+	end
+
+	--- Called any time an ammo crate gets damaged and is about to start burning its ammunition.
+	--- @param Entity entity The affected ammo crate.
+	--- @return boolean # True if the ammo crate can start burning, false otherwise.
+	function Gamemode:ACF_AmmoCanBurn()
+		return true
+	end
+
+	--- Called when an ammo crate attempts to create an explosion, usually due to damage.
+	--- @param Entity entity The affected ammo crate.
+	--- @return boolean # True if the ammo crate can explode, false otherwise.
+	function Gamemode:ACF_AmmoCanExplode()
+		return true
+	end
+
+	--- Called when a fuel tank attempts to replenish another one.
+	--- @param Refill entity The tank that will be providing fuel.
+	--- @param FuelTank entity The tank that will be replenished.
+	--- @param Amount number The quantity of fuel that will get replenished.
+	--- @return boolean # True if the FuelTank can be replenished by the Refill with the given amount, false otherwise.
+	function Gamemode:ACF_FuelCanRefill()
+		return true
+	end
+
+	--- Called when a fuel tank attempts to create an explosion, usually due to damage.
+	--- @param Entity entity The affected fuel tank.
+	--- @return boolean # True if the fuel tank can explode, false otherwise.
+	function Gamemode:ACF_FuelCanExplode()
+		return true
+	end
+
+	--- Called when a weapon attempts to fire a projectile.
+	--- @param Entity entity The weapon attempting to fire.
+	--- @return boolean # True if the weapon can be fired, false otherwise.
+	function Gamemode:ACF_WeaponCanFire()
+		return true
+	end
+
+	--- Called when a player switches between safezones.
+	--- @param Player entity The affected player.
+	--- @param Zone? string | nil The zone which the player moved into, could be nil.
+	--- @param OldZone? string | nil The zone which the player moved from, could be nil.
+	function Gamemode:ACF_PlayerChangedZone()
```

Comment:
Do `ACF_PlayerChangedZone` and `ACF_ProtectionModeChanged` fit the naming standard? They seem kinda out of place.

Reply:
They don't.

Reply:
Updated these hook names to fit the standard